app/federation/server/_client/callbacks/afterSetReaction.js

```diff
@@ -0,0 +1,26 @@
+import _ from 'underscore';
+
+import { FederationRoomEvents, Rooms } from '../../../../models/server';
+import { callbacks } from '../../../../callbacks';
+import { logger } from '../../logger';
+import { Federation } from '../../federation';
+import getFederatedRoomData from './helpers/getFederatedRoomData';
+
+async function afterSetReaction(message, { user, reaction }) {
+	const room = Rooms.findOneById(message.rid);
+
+	// If there are not federated users on this room, ignore it
+	if (!getFederatedRoomData(room).hasFederatedUser) { return; }
+
+	logger.client.debug(`afterSetReaction => message=${ JSON.stringify(_.pick(message, '_id', 'msg'), null, 2) } room=${ JSON.stringify(_.pick(room, '_id'), null, 2) } user=${ JSON.stringify(_.pick(user, 'username'), null, 2) } reaction=${ reaction }`);
```

Comment:
we'll need to get to a better solution for logs like this.. it is doing a bunch of operations (stringify, pick) that start adding taking some time to do under heavy loads